substrate/frame/proxy/src/lib.rs

```diff
@@ -134,20 +148,11 @@ pub mod pallet {
 			+ Default
 			+ MaxEncodedLen;
 
-		/// The base amount of currency needed to reserve for creating a proxy.
-		///
-		/// This is held for an additional storage item whose value size is
-		/// `sizeof(Balance)` bytes and whose key size is `sizeof(AccountId)` bytes.
-		#[pallet::constant]
-		type ProxyDepositBase: Get<BalanceOf<Self>>;
```

Comment:
now we also can remove `Currency` config type parameter, which also depend on old Currency traits. so this is a small win. 
but you also need to adjust the benchmarks, since they use that Currency to ensure the caller has balance for deposits. instead you can use `Consideration::ensure_successful(..)`.

Reply:
Done, benchmarks used currency only so there would be something for gas prices, so I refactored so benchmarks would use balances pallet directly instead of referring to it from `Config`

Reply:
Benchmarks has to be as general as a pallet because you do not know how a client will setup this pallet. If client does not use balances pallet for `Considerations` trait impl, these benchmarks will be wrong. This is why you need to use `Consideration::ensure_successful(..)`.

Reply:
Only tests are in a set runtime (test mocks). Also when you run `cargo test -p pallet-proxy --features=runtime-benchmarks`  the benchmarks run within test runtime, the `impl_benchmark_test_suite` macro does it. But the main target is some runtime setting up this pallet with other impls we do not know.